### PR TITLE
add tof file proxy for data w/o rdh

### PR DESCRIPTION
**Pull request by noferini**
add a file proxy reader to read TOF file acquired in slow control machines (no RDH) 
+ a fix for initOrbit when RDH is missing

### CODE REVIEW

**Review by preghenella — DISMISSED** [review]

File: Detectors/TOF/workflow/src/CompressedDecodingTask.cxx [hunk 1]
```diff
@@ -162,6 +164,11 @@ void CompressedDecodingTask::endOfStream(EndOfStreamContext& ec)
 void CompressedDecodingTask::headerHandler(const CrateHeader_t* crateHeader, const CrateOrbit_t* crateOrbit)
 {
   if (mConetMode) {
+    if (mNCrateOpenTF == 0) {
```
Comment (shahor02) [inline Detectors/TOF/workflow/src/CompressedDecodingTask.cxx:169]: Is it safe to rely on the mNCrateOpenTF, and in general, on the rdhHandler? The delivery of every HBF is not guaranteed, so the 1st HBF RDH bearing the o2::trigger::TF might be missed so the mNCrateOpenTF will not be incremented. 
Also, here: https://github.com/AliceO2Group/AliceO2/blob/5b91ca11bd23384b0c742d6b2f8b3c5292270570/Detectors/TOF/workflow/src/CompressedDecodingTask.cxx#L343-L347
For the `mInitOrbit` you can switch to the `DataHeader::firstTForbit`

File: Detectors/TOF/workflow/src/file-proxy.cxx [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    return;
+  }
+
+  /** read data **/
+  int payload = 0;
+  if (mCONET)
+    payload = readCONET();
+  else
+    payload = readFLP();
+  if (payload == 0) {
```
Comment (shahor02) [inline Detectors/TOF/workflow/src/file-proxy.cxx]: braces

**Review by shahor02 — COMMENTED** [review]
Hi @noferini 
See a few comments below, not sure the current code is robust against dropped HBFs.

File: Detectors/TOF/workflow/src/file-proxy.cxx (hunk 2 above)
Comment (noferini) [reply Detectors/TOF/workflow/src/file-proxy.cxx]: done

File: Detectors/TOF/workflow/src/CompressedDecodingTask.cxx (hunk 1 above)
Comment (noferini) [reply Detectors/TOF/workflow/src/CompressedDecodingTask.cxx:169]: I kept mNCrateOpenTF counters but I removed the check on that for posting data (but the conet mode).
 mInitOrbit = RDHUtils::getHeartBeatOrbit(rdhr); removed as well

**Comment by noferini** [comment]
o2 and macOS checks failing for reasons not related to this PR

**Review by preghenella — APPROVED** [review]

**Review by shahor02 — APPROVED** [review]

**Pull request merged**